packages/contracts/contracts/Devcon6.sol

```diff
@@ -56,10 +67,78 @@ contract Devcon6 is Config, BidModel {
             bidder.amount = msg.value;
             bidder.bidderID = _nextBidderID++;
             _bidders[bidder.bidderID] = msg.sender;
+            _raffleParticipants.push(bidder.bidderID);
         }
         emit NewBid(msg.sender, bidder.bidderID, bidder.amount);
     }
 
+    // auctionWinners should be sorted in descending order
+    function settleAuction(uint256[] calldata auctionWinners)
+        external
+        onlyOwner
+        onlyInState(State.BIDDING_CLOSED)
+    {
+        _settleState = SettleState.AUCTION_SETTLED;
+        uint256 biddersCount = getBiddersCount();
+        if (biddersCount <= _raffleWinnersCount) {
+            return;
+        }
+
+        uint256 auctionParticipantsCount = biddersCount - _raffleWinnersCount;
+        uint256 expectedWinnersLength = _auctionWinnersCount;
+        if (auctionParticipantsCount < expectedWinnersLength) {
+            expectedWinnersLength = auctionParticipantsCount;
+        }
+
+        require(
+            auctionWinners.length == expectedWinnersLength,
+            "Devcon6: invalid auction winners length"
+        );
+
+        for (uint256 i = 0; i < auctionWinners.length; i++) {
+            uint256 bidderID = auctionWinners[i];
+            address bidderAddress = _bidders[bidderID];
+            require(
+                bidderAddress != address(0),
+                "Devcon6: given winner does not exist"
+            );
+
+            _bids[bidderAddress].winType = WinType.AUCTION;
+            removeRaffleParticipant(bidderID - 1);
+        }
+    }
+
+    function removeRaffleParticipant(uint256 index) private {
+        uint256 participantsLength = _raffleParticipants.length;
+        require(
+            index < participantsLength,
+            "Devcon6: invalid raffle participant index"
+        );
+        _raffleParticipants[index] = _raffleParticipants[
+            participantsLength - 1
+        ];
+        _raffleParticipants.pop();
+    }
+
+    function getState() public view returns (State) {
```

Comment:
Something to consider in a following PR: What if owner doesn't call settleAuction and settleRaffle? Can money still be withdrawn after _claimingEndTime?